common/permissions.py

```diff
@@ -1,23 +1,76 @@
 from user.groups import COPYEDITOR_GROUP, EDITOR_GROUP
-from rest_framework.permissions import BasePermission
+from rest_framework.permissions import BasePermission, SAFE_METHODS
 
 from user.models import SluglineUser
 
 
-class IsContributorOrAbove(BasePermission):
+def create_permission(read_perm, write_perm):
+    """
+    Creates a new permission class using read_perm
+    for the readonly permission, and write_perm for the write
+    permission.
+
+    Since DRF uses permission classes and not instances of
+    those classes, this function takes classes as arguments and
+    creates a local class and returns it.
+    """
+
+    # we need instances to call the methods on
+    read_instance = read_perm()
+    write_instance = write_perm()
+
+    class NewPermission(BasePermission):
+        def has_permission(self, request, view):
+            if request.method in SAFE_METHODS:
+                return read_instance.has_permission(request, view)
+            else:
+                return write_instance.has_permission(request, view)
+
+        def has_object_permission(self, request, view, obj):
+            if request.method in SAFE_METHODS:
+                return read_instance.has_object_permission(request, view, obj)
+            else:
+                return write_instance.has_object_permission(request, view, obj)
```

Comment:
Would it be possible to move this class outside of the function, and have the read/write_instance be class members?

Reply:
I don't think that would work, since DRF's viewsets take classes as opposed to instantiated types. To have the read/write perms be class members, we'd have to instantiate the new permission to pass them in, and then DRF wouldn't accept them.

Reply:
Ahh, I see, my apologies. If that's the case, I wonder if it may be more performant in the future to cache permissions classes so we don't keep creating new ones.

Reply:
I'm not too concerned about performance, since (theoretically) this only gets run on startup when we declare viewsets.